tests/by-util/test_whereis.rs

```diff
@@ -0,0 +1,60 @@
+use crate::common::util::TestScenario;
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_basic_lookup() {
+    new_ucmd!()
+        .arg("gcc")
+        .succeeds()
+        .stdout_contains("/usr/bin/gcc");
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_bin_only() {
+    new_ucmd!()
+        .arg("-b")
+        .arg("ping")
+        .succeeds()
+        .stdout_contains("/usr/bin/ping");
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_man_only() {
+    new_ucmd!()
+        .arg("-m")
+        .arg("nmap")
+        .succeeds()
+        .stdout_contains("/usr/share/man/man1/nmap.1.gz");
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+// dig doesn't seem to have any output when passing in the -s flag.
```

Comment:
This comment is incorrect. Both your implementation and the original `whereis` have some output:
```
$ cargo run -q whereis dig -s
dig:
$ whereis dig -s
dig:
```
```suggestion
```